superset/assets/src/explore/exploreUtils.js

```diff
@@ -40,19 +42,31 @@ export function getURIDirectory(formData, endpointType = 'base') {
   return directory;
 }
 
-export function getExploreLongUrl(formData, endpointType) {
+export function getExploreLongUrl(formData, endpointType, allowOverflow = true, extraSearch = {}) {
   if (!formData.datasource) {
     return null;
   }
 
   const uri = new URI('/');
   const directory = getURIDirectory(formData, endpointType);
   const search = uri.search(true);
+  Object.keys(extraSearch).forEach((key) => {
+    search[key] = extraSearch[key];
+  });
   search.form_data = JSON.stringify(formData);
   if (endpointType === 'standalone') {
     search.standalone = 'true';
   }
-  return uri.directory(directory).search(search).toString();
+  const url = uri.directory(directory).search(search).toString();
+  if (!allowOverflow && url.length > MAX_URL_LENGTH) {
+    const minimalFormData = {
```

Comment:
Shouldn't we always want to send the minimalFormData?

Reply:
Nah that's falling back on specifying only the datasource and viz_type. Basically the URL in the browser is shortened but looses most of the state. The shortner still behaves as expected though.